src/test/regress/specs/isolation_ensure_dependency_activate_node.spec

```diff
@@ -0,0 +1,170 @@
+# the test expects to have zero nodes in pg_dist_node at the beginning
+# add single one of the nodes for the purpose of the test
+setup
+{
+    SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+	SELECT nodename, nodeport, isactive FROM master_add_node('localhost', 57637);
+}
+
+# ensure that both nodes exists for the remaining of the isolation tests
+teardown
+{
+    -- schema drops are not cascaded
+    SELECT run_command_on_workers($$DROP SCHEMA IF EXISTS myschema CASCADE;$$);
+    DROP SCHEMA IF EXISTS myschema CASCADE;
+
+    RESET search_path;
+    DROP TABLE IF EXISTS t1 CASCADE;
+    DROP TABLE IF EXISTS t2 CASCADE;
+    DROP TABLE IF EXISTS t3 CASCADE;
+
+	SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-add-worker"
+{
+	SELECT nodename, nodeport, isactive FROM master_add_node('localhost', 57638);
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+# printing in session 1 adds the worker node, this makes we are sure we count the objects
+# on that node as well
+step "s1-print-distributed-objects"
+{
+    SELECT nodename, nodeport, isactive FROM master_add_node('localhost', 57638);
+
+    -- print an overview of all distributed objects
+    SELECT pg_identify_object_as_address(classid, objid, objsubid) FROM citus.pg_dist_object;
+
+    -- print if the schema has been created
+    SELECT count(*) FROM pg_namespace where nspname = 'myschema';
+    SELECT run_command_on_workers($$SELECT count(*) FROM pg_namespace where nspname = 'myschema';$$);
+
+    SELECT master_remove_node('localhost', 57638);
```

Comment:
I think a step called `s1-print-distributed-objects` should not remove node. It is hard to follow the test. 

Can you make it a separate step?

Reply:
It also adds the node at the top.

I agree this is hard to follow, but it mostly a debug step that needs to be able to connect to all nodes. Will add a comment at the top of the step to make this more clear